File: leetcode/easy/1752. Check if Array Is Sorted and Rotated/main.go
package main

func check(nums []int) bool {
	var (
		count = 0
		n = len(nums)
	)

	if n <= 1 {
		return true
	}

	for i := range nums {
		if nums[i] > nums[(i + 1) % n] {
			count++
		}
	}

	return count <= 1
}

File: leetcode/easy/1752. Check if Array Is Sorted and Rotated/main.ts
function check(nums: number[]): boolean {
  let [count, n] = [0, nums.length];

  for (let i = 0; i < n; i++) {
    if (nums[i] > nums[(i + 1) % n]) {
      count++;
    }
  }

  return count <= 1;
};
